storage/innobase/xtrabackup/src/backup_copy.cc

```diff
@@ -2655,6 +2654,10 @@ static void decrypt_decompress_thread_func(datadir_thread_ctxt_t *ctxt) {
       continue;
     }
 
+    if (xtrabackup_encrypt_key) {
+      setenv("XBCRYPT_ENCRYPTION_KEY", xtrabackup_encrypt_key, 1);
+    }
+
```

Comment:
here too, we seem to set after every file, this should be set before the while loop isn't it?